src/render.js

```diff
@@ -1,48 +1,40 @@
-import entries from 'object.entries'
-
-const {hasOwnProperty} = Object.prototype
-const tags = {}
-let prevStyles = {}
+const tags = new Map()
+let prevStyles = new Map()
 
 export default typeof window === 'undefined' ? renderOnServer : renderOnClient
 
 function renderOnServer() {}
 
-function renderOnClient(components) {
-  const styles = {}
-  for (const c of components) {
-    styles[c.props.styleId] = c
```

Comment:
I mean maybe we can trick some engine but I don't know if they all keep object keys in order with `'id:'+styleId`.

If there is any real benefit in switching to an ordered data structure I'd say let's merge.
As I said at the beginning order will never be 100% deterministic since we dedupe styles e.g.:

```jsx
// componentA:
<Button />

// componentB: 
<Header><Button /><Header />
```

still renders:

```
button styles
header styles
```